### PR TITLE
make bool Weather::begin() to return true/false

### PR DESCRIPTION
**Pull request by Guex**
with this change bool Weather::begin() truly returns boolen a value. We can then use it properly for conditional statements like "if (Weather.begin() == true) {..."

### CODE REVIEW

**Comment by tikonen** [comment]
Also not returning from non-void function results in undefined behaviour in C++. This will cause crashes and bugs on some architectures and/or compilers.

**Comment by nseidle** [comment]
We are very late to respond but PRs are always appreciated. Thanks @Guex.

File: src/SparkFun_Si7021_Breakout_Library.cpp [hunk 1]
```diff
@@ -60,15 +60,18 @@
...
   else if(x == 2)
   {
     Serial.println("HTU21D Found");
     //Serial.println(ID_Temp_Hum, HEX);
+    return true;
   }
   else
   	Serial.println("No Devices Detected");
   	//Serial.println(ID_Temp_Hum, HEX);
+	 return false;
```
Comment (marksidell) [inline src/SparkFun_Si7021_Breakout_Library.cpp:74]: I think you need braces around these two indented statements.

Comment (nseidle) [reply src/SparkFun_Si7021_Breakout_Library.cpp:74]: Ugh. Good catch.

Comment (nseidle) [reply src/SparkFun_Si7021_Breakout_Library.cpp:74]: Well, while not pretty, it works as intended. If the IC is not x == 1 or 2, it drops and prints the 'No devices detected'. It then completes the if statement and returns false.

Said differently, if it's not a 0x15 or an 0x32, we don't know what it is and begin correctly returns false.

**Pull request merged**